plugins/kotlin/idea/tests/test/org/jetbrains/kotlin/idea/parameterInfo/AbstractParameterInfoTest.kt

```diff
@@ -68,27 +68,36 @@ abstract class AbstractParameterInfoTest : KotlinLightCodeInsightFixtureTestCase
 
             val context = ShowParameterInfoContext(editor, project, file, editor.caretModel.offset, -1, true)
 
-            val handlers = ShowParameterInfoHandler.getHandlers(project, KotlinLanguage.INSTANCE)!!
-            val handler = handlers.firstOrNull { it.findElementForParameterInfo(context) != null }
-                ?: error("Could not find parameter info handler")
-
-            val mockCreateParameterInfoContext = MockCreateParameterInfoContext(file, myFixture)
-            val parameterOwner = handler.findElementForParameterInfo(mockCreateParameterInfoContext) as PsiElement
+            lateinit var handler: ParameterInfoHandler<PsiElement, Any>
+            lateinit var mockCreateParameterInfoContext: MockCreateParameterInfoContext
+            lateinit var parameterOwner: PsiElement
+            executeOnPooledThreadInReadAction {
+                val handlers = ShowParameterInfoHandler.getHandlers(project, KotlinLanguage.INSTANCE)!!
+                @Suppress("UNCHECKED_CAST")
+                handler = (handlers.firstOrNull { it.findElementForParameterInfo(context) != null }
+                    ?: error("Could not find parameter info handler")) as ParameterInfoHandler<PsiElement, Any>
```

Comment:
This line looks like the `error` call is casted to `ParameterInfoHandler`. :p

Isn't it better to do something like:
```kt
handler = handlers.firstOrNull { ... } as? ParameterInfoHandler<...>
    ?: error(...)
```
Also, with `as?`, we may not need `@Suppress`?

Reply:
`getHandlers` returns a raw type hence there's always need for an unchecked cast; can't get rid of the `@Suppress`. However I do agree it reads better the way you did it, with `error` not looking like it's casted!